### PR TITLE
Switch to Vite for building GUI

### PR DESCRIPTION
**Pull request by touilleMan**
# What has changed ?

- [ ] Does this PR affect the User ?

depend on #4228

### CODE REVIEW

**Review by FirelightFlagboy — COMMENTED** [review]
This PR is really big, can it be splited ?

File: oxidation/client-vite/.gitignore
Comment (FirelightFlagboy) [inline oxidation/client-vite/.gitignore:1]: Some of those values should be in the main gitignore

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: oxidation/client-vite/capacitor.config.ts
Comment (FirelightFlagboy) [inline oxidation/client-vite/capacitor.config.ts:1]: Why this file is the same as `oxidation/client-vite/electron/capacitor.config.ts` ?

File: oxidation/client-vite/electron/electron-builder.config.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        "allowElevation": true,
+        "oneClick": false,
+        "allowToChangeInstallationDirectory": true
+    },
+    "win": {
+        "target": "nsis",
+        "icon": "assets/appIcon.ico"
+    },
+    "mac": {
+        "category": "your.app.category.type",
```
Comment (FirelightFlagboy) [inline oxidation/client-vite/electron/electron-builder.config.json:24]: Do we really want the category to be `your.app.category.type` ?

**Review by FirelightFlagboy — COMMENTED** [review]
Could it be possible to rename `client-vite` to `client` and the current `client` folder to `client-old` ?

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by NicoTuxx — APPROVED** [review]

**Pull request merged**